modules/localstack/src/test/java/org/testcontainers/containers/localstack/SimpleLocalstackS3Test.java

```diff
@@ -29,6 +29,9 @@ public void s3Test() throws IOException {
                 .standard()
                 .withEndpointConfiguration(localstack.getEndpointConfiguration(S3))
                 .withCredentials(localstack.getDefaultCredentialsProvider())
+                //workaround to work with latest localstack due to a bug
+                // https://github.com/localstack/localstack/issues/43
```

Comment:
Hmm, this issue is 2 years old - is this really necessary due to the latest localstack version?

I’m a bit concerned that if we require people to start using this because of our base image upgrade, it is a breaking change...

Reply:
Also, the nip.io endpoint URLs that we generate shouldn’t be adversely affected if any random bucket name gets prefixed onto them... I’m away from a computer right now so can’t verify, but what behaviour are you seeing?

Reply:
When trying to create a bucket with withPathStyleAccessEnabled set to false, the s3 client is making this call PUT http://foo.127.0.0.1.nip.io:34273 and the bucket is not getting created.

If I set withPathStyleAccessEnabled to true, the client is making this call PUT http://127.0.0.1.nip.io:34273/foo and the bucket is created

The nip url is not getting affected as a regular localhost url which fails directly but it looks like the parsing done in Localstack is not going throw.

If you are worried about breaking change, what about just enable the static factory for the Service class by updating  @RequiredArgsConstructor(staticName = "of") which should allow anyone to enable the services they want

Reply:
@rnorth what do you think about making the constructor of Service public, so we can create new Service and enable them on demand

Reply:
> The nip url is not getting affected as a regular localhost url which fails directly but it looks like the parsing done in Localstack is not going throw.

This might actually be fixed by #1422 - let's check if that fixes things, as I'd prefer to use that approach if possible.

Reply:
Hi @rnorth, I applied the changed from #1422 and also the one you suggested in your comment but it doesn't change anything.

I am not sure it's possible to make it work without setting .withPathStyleAccessEnabled(true). 

This is even listed in the Locastack troubleshooting section:

``` 
Accessing local S3 from Java: To avoid domain name resolution issues, you need to enable path style access on your client:

s3.setS3ClientOptions(S3ClientOptions.builder().setPathStyleAccess(true).build());
// There is also an option to do this if you're using any of the client builder classes:
AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard();
builder.withPathStyleAccessEnabled(true);
...